### PR TITLE
update version to add metrics accounts support

### PR DESCRIPTION
**Pull request by 8naama**
this PR closes: #163 

- Upgrade to Client v1.21.0
- Add support for Metrics Accounts API
- Update readme version (to v1.15.0) and it's links

**Note:** a warning from cycode-security will most likely pop as it did at #163 for `datasource_metrics_account` for not returning error directly, the reason we do not return it directly is to avoid terraform `no "id" found in attributes` that are caused when it is returned directly (same in [sub accounts](https://github.com/logzio/terraform-provider-logzio/blob/master/logzio/datasource_subaccount.go#L88-L95)):
```Go
func() error {
	return dataSourceMetricsAccountRead(d, m)
}
```

### CODE REVIEW

**Review by ralongit — APPROVED** [review]
Looks good to me.

File: logzio/datasource_metrics_account.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+}
+
+func dataSourceMetricsAccountReadWrapper(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var err error
+	readErr := retry.Do(
+		func() error {
+			// we use this instead of return dataSourceMetricsAccountRead(d, m) directly
+			// to avoid 'no "id" found in attributes' error
+			err = dataSourceMetricsAccountRead(d, m)
+			if err != nil {
```
[the remark this answers is not in the document]
Comment (ralongit) [reply logzio/datasource_metrics_account.go:60]: @8naama Please make the necessary changes

File: logzio/datasource_metrics_account.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+	if readErr != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func dataSourceMetricsAccountRead(d *schema.ResourceData, m interface{}) error {
+	var client *metrics_accounts.MetricsAccountClient
+	client, _ = metrics_accounts.New(m.(Config).apiToken, m.(Config).baseUrl)
```
Comment (yotamloe) [inline logzio/datasource_metrics_account.go:88]: Why this error is ignored?

File: logzio/resource_metrics_account.go [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func MetricsAccountClient(m interface{}) *metrics_accounts.MetricsAccountClient {
+	var client *metrics_accounts.MetricsAccountClient
+	client, _ = metrics_accounts.New(m.(Config).apiToken, m.(Config).baseUrl)
```
Comment (yotamloe) [inline logzio/resource_metrics_account.go:77]: Why this error is ignored?

File: logzio/datasource_metrics_account_test.go [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+	"os"
+	"regexp"
+	"strconv"
+	"testing"
+)
+
+func TestAccDataSourceMetricsAccount(t *testing.T) {
+	resourceName := "logzio_metrics_account.metrics_account_datasource"
+	dataSourceName := "data.logzio_metrics_account.metrics_account_datasource_by_id"
+	accountId, _ := strconv.ParseInt(os.Getenv(envLogzioAccountId), utils.BASE_10, utils.BITSIZE_64)
```
Comment (yotamloe) [inline logzio/datasource_metrics_account_test.go:16]: why this error is ignored?

File: logzio/datasource_metrics_account_test.go [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+				),
+			},
+		},
+	})
+}
+
+func TestAccDataSourceMetricsAccountByAccountName(t *testing.T) {
+	resourceName := "logzio_metrics_account.metrics_account_datasource"
+	dataSourceName := "data.logzio_metrics_account.metrics_account_datasource_by_id"
+	accountId, _ := strconv.ParseInt(os.Getenv(envLogzioAccountId), utils.BASE_10, utils.BITSIZE_64)
```
Comment (yotamloe) [inline logzio/datasource_metrics_account_test.go:51]: why this error is ignored?

File: logzio/datasource_metrics_account.go [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+					}
+				}
+				return false
+			}),
+		retry.DelayType(retry.BackOffDelay),
+		retry.Attempts(15),
+	)
+
+	if readErr != nil {
+		return diag.FromErr(err)
```
Comment (yotamloe) [inline logzio/datasource_metrics_account.go:80]: The code returs `diag.FromErr(err)` on failure, but `err` is the error from the last retry attempt, not necessarily `readErr`. You should return `diag.FromErr(readErr)` to correctly report the error that caused the read operation

File: logzio/datasource_metrics_account.go [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+	}
+}
+
+func dataSourceMetricsAccountReadWrapper(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var err error
+	readErr := retry.Do(
+		func() error {
+			// we use this instead of return dataSourceMetricsAccountRead(d, m) directly
+			// to avoid 'no "id" found in attributes' error
+			if err = dataSourceMetricsAccountRead(d, m); err != nil {
```
Comment (yotamloe) [inline logzio/datasource_metrics_account.go:59]: you're checking `err != nil` then returning `err` or `nil` based on that. This is redundant as you can directly return `err` :)

**Review by yotamloe — CHANGES_REQUESTED** [review]
Amazing job! please see my comments

File: logzio/datasource_metrics_account.go (hunk 7 above)
Comment (8naama) [reply logzio/datasource_metrics_account.go:59]: when returning it directly it's causing an error from terraform `no "id" found in attributes` for some reason:
``` go
func() error {
	return dataSourceMetricsAccountRead(d, m)
}
```
and when using the syntax as used in [sub accounts](https://github.com/logzio/terraform-provider-logzio/blob/master/logzio/datasource_subaccount.go#L88-L95) it's causing a warning from cycode-security like in #163

**Review by yotamloe — APPROVED** [review]

**Pull request merged**